### PR TITLE
added DominikReichl.KeePass 2.56.0

### PR DESCRIPTION
**Pull request by mluckam**
- [X] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [X] This PR only modifies one (1) manifest
- [X] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [X] Have you tested your manifest locally with `winget install --manifest <path>`?
- [X] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/137901)

### CODE REVIEW

**Comment by stephengillie** [comment]
Hi @mluckam,

This manifest removes Apps and Features entries that are present in previous PR versions. Should these entries also be added to this version?

(Automated response - build 701.)

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

**Comment by mdanish-kh** [comment]
> When updating using wingetcreate, the entries DisplayName and UpgradeCode were not updated.



Unless it's an MSI / MSIX, these registry values cannot be fetched directly from the installer file. One needs to install the package first, and wingetcreate doesn't have the ability to do that

> Any thoughts as to these changes having consequences?

These are additional values that help in package correlation for WinGet. Having incorrect values may have adverse effects. The existing metadata already looks enough for correlation since we have an exact match on ProductCode and Publisher values. Sure, the inno installer the MSI do have a different DisplayName (inno -> `KeePass Password Safe <version>` and msi -> `KeePass <version>`), but I believe the fuzzy matching for PackageName, and having an exact match on ProductCode and Publisher will still provide good correlation. 

The manifest creators don't have the ability yet to update these `AppsAndFeaturesEntries` automatically, so for simplifying package updates / contributions to the package, we have more value in removing these entries entirely (unless someone reports an issue where the package isn't getting matched correctly, then we'll add these back in)

File: manifests/d/DominikReichl/KeePass/2.56.0/DominikReichl.KeePass.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  ProductCode: '{03F9E7B5-495F-4605-BF59-358B2CA33280}'
+  Scope: machine
+- Architecture: x86
+  InstallerType: msi
+  InstallerUrl: https://sourceforge.net/projects/keepass/files/KeePass%202.x/2.56/KeePass-2.56.msi/download
+  InstallerSha256: AF7C504B2B16F6F664B5154921B7F8AA3F44FA593D90F96F858A66F6A4C3F4BC
+  InstallerSwitches:
+    Custom: MSIINSTALLPERUSER=1
+  ProductCode: '{03F9E7B5-495F-4605-BF59-358B2CA33280}'
+  Scope: user
```
[the remark this answers is not in the document]
Comment (mluckam) [reply manifests/d/DominikReichl/KeePass/2.56.0/DominikReichl.KeePass.installer.yaml:42]: I looked at the above issue, and the resolution appears to be removing user scope installs.  User scope installs provide a valuable feature to install updates without requiring permission elevation. Is there a suggestion on how to correct the manifest or msi to address this issue without removing the user scope install?

[the remark this answers is not in the document]
Comment (mdanish-kh) [reply manifests/d/DominikReichl/KeePass/2.56.0/DominikReichl.KeePass.installer.yaml:42]: I agree that it's a useful feature, but unfortunately the only users actually getting the user scope install would be those that don't have an existing installation of KeePass and are using WinGet to install this package. Even then, the users planning to upgrade this package through winget would get the "machine" scope install when they try to upgrade, leaving side-by-side installations on their machine.

See -

https://github.com/microsoft/winget-pkgs/assets/88161975/6d0a761b-a1a0-4c95-a04f-9b47ca8722e0

1. winget detects the current 'user' scope installation as 'machine' scope
2. User upgrading get the machine scope install that requires elevation
3. If the user does proceed with the install, user is left with a side-by-side installation, becoming another source of confusion


image


> Is there a suggestion on how to correct the manifest or msi to address this issue without removing the user scope install?

Unless there is a resolution from WinGet's side (tracked in linked issue above) to handle these situations better, or MSIs start putting the registry entries in the correct place, I don't know of a better solution than just removing the user scope from the manifest.

[the remark this answers is not in the document]
Comment (mdanish-kh) [reply manifests/d/DominikReichl/KeePass/2.56.0/DominikReichl.KeePass.installer.yaml:42]: Technical users can still retain the user scope install if they want by passing the args to the installer on update:

```
winget upgrade DominikReichl.KeePass --custom "MSIINSTALLPERUSER=1"
```
But the manifest should exhibit a behavior that causes the least issues for majority users.

[the remark this answers is not in the document]
Comment (mluckam) [reply manifests/d/DominikReichl/KeePass/2.56.0/DominikReichl.KeePass.installer.yaml:42]: > MSIs start putting the registry entries in the correct place
Is this problem MSIs as a whole or misuse per MSI.  If this issue could be corrected in the KeePass MSI, I can reach out to the maintainer.

[the remark this answers is not in the document]
Comment (mdanish-kh) [reply manifests/d/DominikReichl/KeePass/2.56.0/DominikReichl.KeePass.installer.yaml:42]: Reading the linked issue, and reviewing many MSIs PRs, I'm under the impression this is a problem with MSIs (or WiX - a popular toolset to create MSIs?) as a whole. I've yet to encounter an MSI package that does this correctly.

**Comment by stephengillie** [comment]
Duplicate of https://github.com/microsoft/winget-pkgs/pull/138277

**Pull request closed without merging**